apps/sales/serializers.py

```
@@ -0,0 +1,72 @@
+    def create(self, validated_data):
+        items_data = validated_data.pop('items')
+        merchant = self.context['request'].user
+        
+        # 1. Fetch actual Product objects to pass to Service
+        # This creates a cleaner list of {product: obj, quantity: 5}
+        processed_items = []
+        for item in items_data:
+            try:
+                product = Product.objects.get(pk=item['product_id'], merchant=merchant)
+            except Product.DoesNotExist:
+                raise serializers.ValidationError({"items": f"Product {item['product_id']} not found."})
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)